data_structures/linked_list/cycle_in_linked_list.c

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file
+ * @author [Ankita Kulkarni](https://github.com/ankitakul)
+ * @brief This Algorithm detects a loop in a singly linked list 
+*/
+
+#include <stdio.h> // For IO operations (printf)
+#include <stdlib.h> // For the malloc operation
+ 
+/**
+* @brief Linked list node structure
+*/
+struct node {
+  int data; //data assigned to the data part of the node
+  struct node *next; //pointer to next node in linked list
+} *head;
+ 
+ 
+/**
+ * Function insert
+ * @brief This function creates a new node and inserts it in front of a singly linked list
+ * @param num data that is assigned to the node
+ * @returns void
+*/
+
+void insert(int num) {
+    //create a new node
+    struct node* newNode = (struct node*) malloc(sizeof(struct node));
+    
+    //assign value(num)
+    newNode->data  = num;
+    
+    //next pointer of new node will point to head node of linked list
+    newNode->next = head;
+    
+    //head of the list is the new node
+    head = newNode;
+    
+    //printf("Inserted Element : %d\n", num);
```

Comment:
I think this should be used.
```suggestion
    printf("Inserted Element : %d\n", num);
```